src/core/qgspallabeling.cpp

```diff
@@ -3944,7 +3944,7 @@ QgsGeometry* QgsPalLabeling::prepareGeometry( const QgsGeometry* geometry, QgsRe
     clonedGeometry.reset( geom );
   }
 
-  if ( clipGeometry && !clipGeometry->contains( geom ) )
+  if ( clipGeometry && !clipGeometry->boundingBox().contains( geom->boundingBox() ) )
```

Comment:
@ahuarte47 Sorry, I don't quite understand. The changes to geometryRequiresPreparation are fine, no issue there. But we need to make sure the test before doing the actual clipping is still applied using the geometries themselves when rotation is applied. Maybe a better replacement for line 3947 would be:

      if ( clipGeometry &&
      ( ( qgsDoubleNear( m2p.mapRotation(), 0 ) && !clipGeometry->boundingBox().contains( geom->boundingBox() ) )
        || ( !qgsDoubleNear( m2p.mapRotation(), 0 ) && !clipGeometry->contains( geom ) ) ) )

Ie if no rotation, do the optimised bounding box check, but if rotation then do the full expensive check?